### PR TITLE
feat(atomic): add search-box v1 interactivity + accessibility

### PR DESCRIPTION
**Pull request by ThibodeauJF**
https://coveord.atlassian.net/browse/KIT-851

### CODE REVIEW

File: packages/atomic/src/components/atomic-search-box-v1/atomic-search-box.tsx [hunk 1]
```diff
@@ -46,60 +67,242 @@ export class AtomicSearchBox {
...
+      case 'Enter':
+        this.onSubmit();
+        break;
+      case 'Escape':
+        this.onBlur();
+        break;
+    }
+  }
+
+  private updateQuery(query: string) {
```
Comment (ThibodeauJF) [inline packages/atomic/src/components/atomic-search-box-v1/atomic-search-box.tsx:179]: Works pretty well for our use case! We can always add a method on the control so updating without fetching new query suggest is possible :D

File: packages/atomic/src/components/atomic-search-box-v1/atomic-search-box.tsx [hunk 2]
```diff
@@ -46,60 +67,242 @@ export class AtomicSearchBox {
...
+      this.activeDescendantElement.click();
+      this.updateActiveDescendant();
       return;
     }
 
+    this.searchBox.submit();
+    this.updateActiveDescendant();
+  }
+
+  private onKeyup(e: KeyboardEvent) {
```
Comment (olamothe) [inline packages/atomic/src/components/atomic-search-box-v1/atomic-search-box.tsx:168]: Just wondering why you used `keyUp` to handle submit and clear, and `keyDown` for navigation. Why not the same event for both ?

File: packages/atomic/src/components/atomic-search-box-v1/atomic-search-box.tsx [hunk 3]
```diff
@@ -46,60 +67,242 @@ export class AtomicSearchBox {
...
+        aria-haspopup="true"
+        aria-owns={this.popupId}
+        aria-expanded={`${this.isExpanded}`}
+        aria-activedescendant={this.activeDescendant}
+        autocomplete="off"
+        autocapitalize="off"
+        autocorrect="off"
+        placeholder={this.bindings.i18n.t('search')}
+        aria-label={this.bindings.i18n.t('search-box')}
+        type="text"
```
Comment (olamothe) [inline packages/atomic/src/components/atomic-search-box-v1/atomic-search-box.tsx:217]: Should it be `type=search` ?

File: packages/atomic/src/components/atomic-search-box-v1/atomic-search-box.tsx [hunk 4]
```diff
@@ -46,60 +67,242 @@ export class AtomicSearchBox {
...
+        break;
+      case 'Escape':
+        this.onBlur();
+        break;
+    }
+  }
+
+  private updateQuery(query: string) {
+    this.bindings.engine.dispatch(
+      loadQuerySetActions(this.bindings.engine).updateQuerySetQuery({
```
Comment (olamothe) [inline packages/atomic/src/components/atomic-search-box-v1/atomic-search-box.tsx:181]: Due to the way that these `loadActions` function are currently coded in headless, they always execute redux specific code to `replaceReducers`. 

An (easy) optimization would be to just load it once in `initialize`.

**Review by olamothe — APPROVED** [review]

File: packages/atomic/src/components/atomic-search-box-v1/atomic-search-box.tsx (hunk 3 above)
Comment (ThibodeauJF) [reply packages/atomic/src/components/atomic-search-box-v1/atomic-search-box.tsx:217]: no type search add custom stuff we don't want https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/search

**Comment by ThibodeauJF** [comment]
https://github.com/coveo/ui-kit/pull/1184 will still be opened!

**Pull request merged**